### ISSUE TITLE
RockOn Env UI needs scroll bar

### ISSUE THREAD

**Issue by superheroham** [rockstor/rockstor-core#2598]
I have been working on writing a RockOn for "PhotoPrism".  There are nearly 30 env entries in the Composer file they provide as an example.  The problem is the RockStor UI doesn't have a way to scroll to see them all during installation...

```swift
{
	"PhotoPrism": {
		"description": "AI-Powered Photos App for the Decentralized Web",
		"version": "0.1",
		"website": "https://www.photoprism.app/",
		"icon": "https://www.photoprism.app/favicon.ico",
		"more_info": "",
		"ui": {
			"slug": "gui"
		},
		"volume_add_support": true,
		"containers": {
			"photo prism": {
				"image": "photoprism/photoprism",
				"launch_order": "1",
				"ports": {
					"2342": {
						"description": "PhotoPrism UI Port",
						"label": "Web-UI",
						"host_default": 2342,
						"protocol": "tcp",
						"ui": true
					}
				},
				"environment": {
					"PHOTOPRISM_ADMIN_USER": {
						"description": "Default WebUI Username",
						"label": "Admin Username",
						"index": 1,
						"default_value": "admin"
					},
					"PHOTOPRISM_ADMIN_PASSWORD": {
						"description": "initial admin password (8-72 characters)",
						"label": "Admin Password",
						"index": 2,
						"default_value": "insecure"
					},
					"HOTOPRISM_AUTH_MODE": {
						"description": "authentication mode (public, password)",
						"label": "Auth Mode",
						"index": 3,
						"default_value": "password"
					},
					"PHOTOPRISM_SITE_URL": {
						"description": "server URL in the format http(s): //domain.name(:port)/(path)",
						"label": "Site URL",
						"index": 4,
						"default_value": "http://photoprism.me:2342/"
					},
					"PHOTOPRISM_ORIGINALS_LIMIT": {
						"description": "file size limit for originals in MB (increase for high-res video)",
						"label": "Orginals Size Limit (MB)",
						"index": 5,
						"default_value": "5000"
					},
					"PHOTOPRISM_HTTP_COMPRESSION": {
						"description": "improves transfer speed and bandwidth utilization (none or gzip)",
						"label": "HTTP Compression",
						"index": 6,
						"Default_value": "gzip"
					},
					"PHOTOPRISM_LOG_LEVEL": {
						"description": "log level: trace, debug, info, warning, error, fatal, or panic",
						"label": "Log Level",
						"index": 7,
						"default_value": "info"
					},
					"PHOTOPRISM_READONLY": {
						"description": "do not modify originals directory (reduced functionality)",
						"label": "Read Only (true/false)",
						"index": 8,
						"default_value": "false "
					},
					"PHOTOPRISM_EXPERIMENTAL": {
						"description": "enables experimental features",
						"label": "Experimental Features (true/false)",
						"index": 9,
						"defaut_value": "false"
					},
					"PHOTOPRISM_DISABLE_CHOWN": {
						"description": "disables updating storage permissions via chmod and chown on startup",
						"label": "Disable CHOWN (true/false)",
						"index": 10,
						"default_value": "false"
					},
					"PHOTOPRISM_DISABLE_WEBDAV": {
						"description": "disables built-in WebDAV server",
						"label": "Disable WEBDAV (true/False)",
						"index": 11,
						"default_value": "false"
					},
					"PHOTOPRISM_DISABLE_SETTINGS": {
						"description": "disables settings UI and API",
						"label": "Disable Settings (true/false)",
						"index": 12,
						"default_value": "false"
					},
					"PHOTOPRISM_DISABLE_TENSORFLOW": {
						"description": "disables all features depending on TensorFlow",
						"label": "Disable TensorFlow (true/false)",
						"index": 13,
						"default_value": "false"
					},
					"PHOTOPRISM_DISABLE_FACES": {
						"description": "disables face detection and recognition (requires TensorFlow)",
						"label": "Disable Face Recognition (true/false)",
						"index": 14,
						"default_value": "false"
					},
					"PHOTOPRISM_DISABLE_CLASSIFICATION": {
						"description": "disables image classification (requires TensorFlow)",
						"label": "Disable Classification (true/false)",
						"index": 15,
						"default_value": "false"
					},
					"PHOTOPRISM_DISABLE_VECTORS": {
						"description": "disables vector graphics support",
						"label": "Disable Vector Formats (true/false)",
						"index": 16,
						"default_value": "false"
					},
					"PHOTOPRISM_DISABLE_RAW": {
						"description": "disables indexing and conversion of RAW images",
						"label": "Disable RAW Format (true/false)",
						"index": 17,
						"default_value": "false"
					},
					"PHOTOPRISM_RAW_PRESETS": {
						"description": "enables applying user presets when converting RAW images (reduces performance)",
						"label": "Use RAW User Presets (true/false)",
						"index": 18,
						"default_value": "false"
					},
					"PHOTOPRISM_JPEG_QUALITY": {
						"description": "a higher value increases the quality and file size of JPEG images and thumbnails (25-100)",
						"label": "JPEG Quality (25-100)",
						"index": 19,
						"default_value": 85
					},
					"PHOTOPRISM_DETECT_NSFW": {
						"description": "automatically flags photos as private that MAY be offensive (requires TensorFlow)",
						"label": "Detect Private/Non-Safe (true/false)",
						"index": 20,
						"default_value": "false"
					},
					"PHOTOPRISM_UPLOAD_NSFW": {
						"description": "allows uploads that MAY be offensive (no effect without TensorFlow)",
						"label": "Allow Upload Private/Non-Safe (true/false)",
						"index": 21,
						"default_value": "true"
					},
					"PHOTOPRISM_DATABASE_DRIVER": {
						"description": "Database Engine to use (sqlite or mysql) MySql supports MariaDB 10.5+ or MySQL 8+",
						"label": "Database Engine (sqlite/mysql)",
						"index": 22,
						"default_value": "mysql"
					},
					"PHOTOPRISM_DATABASE_SERVER": {
						"description": "MariaDB or MySQL database server (hostname:port)",
						"label": "MariaDB/MySQL Server (hostname:port)",
						"index": 23,
						"default_value": "mariadb:3306"
					},
					"PHOTOPRISM_DATABASE_NAME": {
						"description": "MariaDB or MySQL database name",
						"label": "MariaDB/MySQL Database Name",
						"index": 24,
						"default_value": "photoprism"
					},
					"PHOTOPRISM_DATABASE_USER": {
						"description": "MariaDB or MySQL database user name",
						"label": "MariaDB/MySQL User Name",
						"index": 25,
						"default_value": "photoprism"
					},
					"PHOTOPRISM_DATABASE_PASSWORD": {
						"description": "MariaDB or MySQL database user password",
						"label": "MariaDB/MySQL Password",
						"index": 26,
						"default_value": "insecure"
					},
					"PHOTOPRISM_SITE_CAPTION": {
						"description": "Site Title/Caption",
						"label": "Site Caption",
						"index": 27,
						"default_value": "AI-Powered Photos App"
					},
					"PHOTOPRISM_SITE_DESCRIPTION": {
						"description": "Site Meta-Data Description",
						"label": "Meta-Data Descriptions",
						"index": 28
					},
					"PHOTOPRISM_SITE_AUTHOR": {
						"description": "Site Meta-Data Author",
						"label": "Meta-Data Site Author",
						"index": 29
					}
				}

				,
				"volumes": {
					"/photoprism/storage": {
						"description": "Writable* storage folder for cache, database, and sidecar files (DO NOT REMOVE)",
						"label": "PhotoPrism Config/Cache/DB"
					},
					"/photoprism/originals": {
						"description": "Original media files (DO NOT REMOVE)",
						"label": "Originals Folder"
					},
					"/photoprism/import": {
						"description": "base folder from which files can be imported to originals",
						"label": "Imports Folder"
					}
				}

			}
		}
	}
}
```

**Comment by phillxnet** [comment]
@superheroham Hello there.

Does the following not address your issue:
"adds scrollbar on large env vars forms #2533" #2530 by @kanecko 

That was released in testing version 4.5.9-1 which was our sixth stable release candidate:
https://forum.rockstor.com/t/v4-5-testing-channel-changelog/8546/12

Our latest release however is one on from that at 4.6.0-0:
https://forum.rockstor.com/t/v4-5-testing-channel-changelog/8546/13
and is our proposed next stable release, but due to some external changes, we have brought forward the dropping of our legacy shop, so we may just pop in some changed instructions re Stable updates activation that are backed into the Web-UI.

So give 4.6.0-0  (Seventh Stable Release Candidate) form testing a try, if it hasn't already been promoted to stable (planned soon), but avoid anything newer in testing unless it is also indicated as a newer Stable release candidate as we are about to begin a new testing phase which involves quite a lot of changes that will inevitably lead to unknown and new bugs. The plan is to number the new testing phase releases from 5.0.0-0 onwards.

If 4.6.0-0 works for you, can you indicate the same in this issue. As it very much look like the referenced pull request does exactly what you have detailed here as a failing.

**Comment by superheroham** [comment]
I am on latest stable 4.5.8.  I will move to 4.6 and try again.

Sent from my iPhone

On Jul 5, 2023, at 2:46 AM, Philip Guyton ***@***.***> wrote:

﻿

@superheroham<https://github.com/superheroham> Hello there.

Does the following not address your issue:
"adds scrollbar on large env vars forms #2533<https://github.com/rockstor/rockstor-core/issues/2533>" #2530<https://github.com/rockstor/rockstor-core/pull/2530> by @kanecko<https://github.com/kanecko>

That was released in testing version 4.5.9-1 which was our sixth stable release candidate:
https://forum.rockstor.com/t/v4-5-testing-channel-changelog/8546/12

Our latest release however is one on from that at 4.6.0-0:
https://forum.rockstor.com/t/v4-5-testing-channel-changelog/8546/13
and is our proposed next stable release, but due to some external changes, we have brought forward the dropping of our legacy shop, so we may just pop in some changed instructions re Stable updates activation that are backed into the Web-UI.

So give 4.6.0-0 (Seventh Stable Release Candidate) form testing a try, if it hasn't already been promoted to stable (planned soon), but avoid anything newer in testing unless it is also indicated as a newer Stable release candidate as we are about to begin a new testing phase which involves quite a lot of changes that will inevitably lead to unknown and new bugs. The plan is to number the new testing phase releases from 5.0.0-0 onwards.

If 4.6.0-0 works for you, can you indicate the same in this issue. As it very much look like the referenced pull request does exactly what you have detailed here as a failing.

**Comment by superheroham** [comment]
Fixed!  Upgrading to this testing release did the trick!  Thank you!

From: John Hamilton ***@***.***>
Date: Wednesday, July 5, 2023 at 8:55 AM
To: rockstor/rockstor-core ***@***.***>
Subject: Re: [rockstor/rockstor-core] RockOn Env UI needs scroll bar (Issue #2598)
I am on latest stable 4.5.8.  I will move to 4.6 and try again.
Sent from my iPhone


On Jul 5, 2023, at 2:46 AM, Philip Guyton ***@***.***> wrote:
﻿

@superheroham<https://github.com/superheroham> Hello there.

Does the following not address your issue:
"adds scrollbar on large env vars forms #2533<https://github.com/rockstor/rockstor-core/issues/2533>" #2530<https://github.com/rockstor/rockstor-core/pull/2530> by @kanecko<https://github.com/kanecko>

That was released in testing version 4.5.9-1 which was our sixth stable release candidate:
https://forum.rockstor.com/t/v4-5-testing-channel-changelog/8546/12

Our latest release however is one on from that at 4.6.0-0:
https://forum.rockstor.com/t/v4-5-testing-channel-changelog/8546/13
and is our proposed next stable release, but due to some external changes, we have brought forward the dropping of our legacy shop, so we may just pop in some changed instructions re Stable updates activation that are backed into the Web-UI.

So give 4.6.0-0 (Seventh Stable Release Candidate) form testing a try, if it hasn't already been promoted to stable (planned soon), but avoid anything newer in testing unless it is also indicated as a newer Stable release candidate as we are about to begin a new testing phase which involves quite a lot of changes that will inevitably lead to unknown and new bugs. The plan is to number the new testing phase releases from 5.0.0-0 onwards.

If 4.6.0-0 works for you, can you indicate the same in this issue. As it very much look like the referenced pull request does exactly what you have detailed here as a failing.

**Comment by phillxnet** [comment]
@superheroham Thanks for the confirmation.

I'll go ahead and close this issue as a duplicate of @kanecko then.

Re:

> I am on latest stable 4.5.8.

4.5.8-0 was actually our Stable Release Candidate (RC5), we used it in our downloadable installers to get folks at least closer to what we are about to release as our stable rpm: given it had been so long since our last 4.1.0-0 proper stable release. We intended to make our next testing phase much shorter to hopefuly avoid the long delay experience last time. This delay was, in a major part, down to us having to completely change-out our build system. But we are in a far better position as a result.


Closing as
Duplicate of #2533 
Fixed by #2530 by @kanecko
